qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -419,6 +623,79 @@ def _get_sweep_data(self, force_polar: bool = False) -> np.ndarray:
                 self.root_instrument.cont_meas_on()
         return data
 
+    def setup_cw_sweep(self) -> None:
+        """
+        Sets CW mode.
+        """
+
+        # set the channel type to single point msmt
+        self.sweep_type('CW_Point')
+        # turn off average on the VNA since we want single point sweeps.
+        self.write(f'SENS{self._instrument_channel}:AVER:STAT OFF')
+        # This format is required for getting both real and imaginary parts.
+        self.format('Complex')
+        # Set the sweep time to auto such that it sets the delay to zero
+        # between each point (e.g msmt speed is optimized). Note that if one
+        # would like to do a time sweep with time > npts/bandwidth, this is
+        # where the delay would be set, but in general we want to measure as
+        # fast as possible without artificial delays.
+        self.write(f'SENS{self._instrument_channel}:SWE:TIME:AUTO ON')
+        # Set cont measurement off here so we don't have to send that command
+        # while measuring later.
+        self.root_instrument.cont_meas_off()
+
+    def setup_lin_sweep(self) -> None:
+        """
+        Function in order to revert setup_CW_sweep and go back to lin sweep
+        mode.
```

Comment:
the docstring of this method does not need to know about "reverting" to some other mode etc etc - just keep it simple to what it does and that's it :) it will be less effort to maintain and understand. smth like this:
```suggestion
        Setup the instrument into linear sweep mode.
```